File: sion_fasion_shop/ProyectoMVC/SionFashioWebApplicationMVC/SionFashioWebApplicationMVC/Models/categoria.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace SionFashioWebApplicationMVC.Models;

public partial class categoria
{
    [Key]
    public int id_categoria { get; set; }

    [Required(ErrorMessage = "El nombre de la categoría es obligatorio.")]
    [StringLength(100, ErrorMessage = "El nombre de la categoría no puede exceder los 100 caracteres.")]
    public string nombre_categoria { get; set; } = null!;

    [InverseProperty("id_categoriaNavigation")]
    public virtual ICollection<sub_categoria> sub_categoria { get; set; } = new List<sub_categoria>();
}

File: sion_fasion_shop/Entregables/Trimestre V/Sistema de inventario en Node.Js/Bck-End SionFashion-Shop/models/categoria.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const categoriaSchema = new Schema({
    
    nombre_categoria: {
        type: String,
        required: [true, 'El nombre de la categoría es obligatorio.'],
        maxlength: [100, 'El nombre de la categoría no puede exceder los 100 caracteres.']
    },
    sub_categoria: [{
        type: Schema.Types.ObjectId,
        ref: 'SubCategoria' // Asegúrate de que este sea el nombre correcto del modelo sub_categoria
    }]
});

// Exportar el modelo
module.exports = mongoose.model('Categoria', categoriaSchema);
